### PR TITLE
Add path filters to the TPU job

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes https://github.com/Lightning-AI/lightning/issues/14484
Uses https://github.com/marketplace/actions/trigger-circleci-pipeline

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @carmocca @akihironitta @borda

### CODE REVIEW

File: .github/workflows/ci-circleci.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      - "setup.cfg"  # includes pytest config
+
+jobs:
+  # https://circleci.com/blog/trigger-circleci-pipeline-github-action
+  trigger-circleci:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: circleci/trigger_circleci_pipeline@v1.0
+        env:
+          CCI_TOKEN: ${{ secrets.CCI_TOKEN }}
```
Comment (carmocca) [inline .github/workflows/ci-circleci.yml:19]: @Borda I need this token. Do you know how to get it?

info: https://circleci.com/blog/trigger-circleci-pipeline-github-action

File: .circleci/config.yml [hunk 2]
```diff
@@ -4,33 +4,10 @@ orbs:
...
-      - '*'
-  branches:
-    include:
-      - "master"
-      - "release/*"
-      - "refs/tags/*"
-
-pr:
-  - "master"
-  - "release/*"
```
Comment (carmocca) [inline .circleci/config.yml:20]: `.github/workflows/ci-circleci.yml` is what triggers this job now, so this shouldn't be required

**Review by awaelchli — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

File: .github/workflows/ci-circleci.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      - "src/pytorch_lightning/**"
+      - "tests/tests_pytorch/**"
+      - "setup.cfg"  # includes pytest config
+  pull_request:
+    branches: [master, "release/*"]
+    paths:
+      - ".github/workflows/ci-circleci.yml"
+      - ".circleci/config.yml"
+      - "requirements/pytorch/**"
+      - "src/pytorch_lightning/**"
```
Comment (carmocca) [inline .github/workflows/ci-circleci.yml:17]: fyi: lite will need to be added here eventually @awaelchli

**Pull request merged**